### PR TITLE
Add `vessel install --force`

### LINKED ISSUE #55: Force/sync updates when following a branch or version tag
## Motivation
Currently, vessel allows dependency tracking through a commit hash, a tag, or a branch name.

The code for both tags and branch names of a dependency can be updated remotely by the package developer. When this happens, a `vessel install` doesn't automatically fetch the newest version of the branch/tag code from GitHub.

See this [topic issue and discussion](https://forum.dfinity.org/t/updating-vessel-package-when-version-points-to-a-git-branch/14641) from the developer forums

## Potential Solution(s)

1. Add a -f flag, which would force re-fetch all of dependencies of a project listed in the package-set.dhall, and overwrite any dependencies that already exist. This would be very much similar to npm install <dependency> --force, as shown in the docs for [npm install](https://docs.npmjs.com/cli/v6/commands/npm-install).

2. An alternative solution to this would be to include a `release` property that could be used in place of the version in the `package-set.dhall` if one wanted to specifically track a branch/tag in which the underlying code is update-able, which would signify that this package should be re-fetched on every vessel install.

@ggreif @crusso

### PR DESCRIPTION
**Pull request by rvanasa**
Resolves #55 by adding an optional `--force` / `-f` flag to `vessel install`.

Since this feature requires behavior equivalent to `rm -rf`, this PR also fixes a potentially major vulnerability in the form of using unsanitized package names or versions containing `..`, `/`, or `~`, making it possible to modify (or delete) any directory on a user's machine. I fixed this by adding basic package name/version validation with the following rules:

- Package name/version must match the following regex: `[A-Za-z0-9.-_]+` (i.e. the [Portable Filename Character Set](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_282))
- Package name/version cannot consist entirely of the `.` character
- Package name/version cannot start with a `-` character

This validation could break backwards-compatibility with some unusually-named packages, but this seems to be worth it for security reasons.

### CODE REVIEW

**Review by crusso — APPROVED** [review]
LGTM, but my be worth getting a Rust programmer to review too (@chenyan-dfinity or @kentosugama ?)

File: src/lib.rs [hunk 1]
```diff
@@ -198,9 +199,29 @@ impl Vessel {
     }
 }
 
+/// Guards against path strings in package data
+fn is_clean(input: &str) -> bool {
```
Comment (kentosugama) [inline src/lib.rs:203]: Would it make sense to use something like this:
https://docs.rs/sanitize-filename/0.4.0/sanitize_filename/index.html

I'm always paranoid when I see a hand rolled sanitizer

**Comment by kentosugama** [comment]
Curious about your reasoning for using option 1 instead of option 2 from #55

**Comment by rvanasa** [comment]
Adding a flag to the existing `install` command seems like an elegant solution, especially since this is already a well-established feature in NPM and Cargo. 

We could also add something like a `release` property in the future if this would be useful as well.

File: src/lib.rs (hunk 1 above)
Comment (rvanasa) [reply src/lib.rs]: I tried using both [path-clean](https://docs.rs/path-clean/latest/path_clean) and [sanitize-filename](https://crates.io/crates/sanitize-filename), but they both allow various path-like filenames such as `~`, which in our case makes it possible to replace someone's home directory with a Vessel package. IMO, this makes it worth using simple, explicit [POSIX-style](https://www.gnu.org/software/automake/manual/html_node/Limitations-on-File-Names.html) validation (at least for now).

**Comment by rvanasa** [comment]
Would someone mind merging this while @kentosugama and I are waiting to get review permissions for this repository? Thanks!

**Pull request merged**